src/Tests/Microsoft.NET.Sdk.WorkloadManifestReader.Tests/SdkDirectoryWorkloadManifestProviderTests.cs

```diff
@@ -293,6 +279,82 @@ var sdkDirectoryWorkloadManifestProvider
                 .BeEquivalentTo("6.0.100", "5.0.100");
         }
 
+        [Fact]
+        public void ItShouldFallbackWhenPreviewFeatureBandHasNoManifests()
+        {
+            var testDirectory = _testAssetsManager.CreateTestDirectory().Path;
+            var fakeDotnetRootDirectory = Path.Combine(testDirectory, "dotnet");
+
+            // Write 4.0.100 manifests-> android only
+            CreateMockManifest(fakeDotnetRootDirectory, "4.0.100", "iOS");
+
+            // Write 5.0.100 manifests-> android
+            CreateMockManifest(fakeDotnetRootDirectory, "5.0.100", "Android");
+
+            // Write 6.0.100-preview.2 manifests-> ios only
+            CreateMockManifest(fakeDotnetRootDirectory, "6.0.100-preview.2", "iOS");
+
+            // Write 7.0.100 manifests-> ios and android
+            CreateMockManifest(fakeDotnetRootDirectory, "7.0.100", "Android");
+            CreateMockManifest(fakeDotnetRootDirectory, "7.0.100", "iOS");
+
+            var knownWorkloadsFilePath = Path.Combine(fakeDotnetRootDirectory, "sdk", "6.0.100-preview.4.12345", "IncludedWorkloadManifests.txt");
+            Directory.CreateDirectory(Path.GetDirectoryName(knownWorkloadsFilePath)!);
+            File.WriteAllText(knownWorkloadsFilePath, "Android\niOS");
+
+            var sdkDirectoryWorkloadManifestProvider
+                = new SdkDirectoryWorkloadManifestProvider(sdkRootPath: fakeDotnetRootDirectory, sdkVersion: "6.0.100-preview.4.12345", userProfileDir: null);
+
+            GetManifestContents(sdkDirectoryWorkloadManifestProvider)
+                .Should()
+                .BeEquivalentTo("4.0.100", "5.0.100");
```

Comment:
How about including both the manifest ID and version band in the mock manifest contents so you can verify which manifest came from which version?  IE:

```suggestion
                .BeEquivalentTo("4.0.100/iOS", "5.0.100/Android");
```